mupen64plus-core/src/pi/pi_controller.c

```diff
@@ -56,6 +58,11 @@ static void dma_pi_read(struct pi_controller *pi)
    uint32_t rom_address;
    const uint8_t* dram;
    uint8_t* rom;
+   
+   if( !AllowUnalignedDMA ) {
+	   pi->regs[PI_CART_ADDR_REG] &= 0xfffffffe;
+	   pi->regs[PI_DRAM_ADDR_REG] &= 0xfffffffe;
```

Comment:
I initially had the RAM address aligned to the doubleword, but it looks like this was changed in the mupen code to only be halfword aligned, so I'm following that.